src/Migrations.ts

```diff
@@ -0,0 +1,207 @@
+import * as path from 'node:path'
+import * as fsp from 'node:fs/promises'
+
+import { Database } from './Database'
+import { IMigrate } from './interfaces'
+
+import MigrationFile = IMigrate.MigrationFile
+import MigrationData = IMigrate.MigrationData
+
+const DEFAULT_MIGRATIONS_DIR_NAME = 'migrations'
+const DEFAULT_MIGRATIONS_DIR_PATH = path.join(
+  process.cwd(),
+  DEFAULT_MIGRATIONS_DIR_NAME
+)
+
+const MIGRATION_FILE_REGEXP = /^(\d+).(.*?)\.sql$/
+const MIGRATION_FILE_UP_REGEXP = /^-- .*$/gm
+const MIGRATION_FILE_DOWN_REGEXP = /^--\s+?down\b/im
+const MIGRATION_FILE_REPLACE_COMMAND_REGEXP = /^-{3,}$/gim
+
+export class Migrations<
+  Driver extends Database,
+  Config extends IMigrate.MigrationParams = IMigrate.MigrationParams
+> {
+  db: Driver
+  config: Config
+
+  constructor (db: Driver, config?: Config) {
+    this.db = db
+    /** Sequence in the properties definitions is IMPORTANT */
```

Comment:
it's not important. if it is then you're doing something wrong

in JS, object property ordering isn't guaranteed

Reply:
I think I understand what you mean here - you need the default on the `migrationsPath`. You don't need the comment for that though.

Reply:
That's why I wrote `sequence` is important. I don't know how to say it correctly and easy for others to understand. Maybe you can help write a comment if we need one